src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/TestClasses.cs

```diff
@@ -158,4 +158,52 @@ public struct PersonStruct
         public string FirstName { get; set; }
         public string LastName { get; set; }
     }
+
+
+
+    public class CovariantBase
```

Comment:
I'll suggest also to add cases similar to the one below:
- CovariantBaseNotIgnored
- CovariantBaseIgnored

and multiple cases for each variant

Reply:
Done.